.github/workflows/deploy.yml

```diff
@@ -35,7 +36,7 @@ jobs:
         run: make build
 
       - name: Publish to PyPI
-        if: github.event_name == 'push' && startsWith(github.ref, 'refs/heads/master')
+        if: github.event.pull_request.merged == true && startsWith(github.ref, 'refs/heads/master')
```

Comment:
I'd suggest publishing through a release process instead of doing it on every merge to `master` so that there is more control and it's more intentional when a new version of the SDK is available. 💭

Reply:
Would love some help with this part actually on your down time of course